druid-shell/src/platform/x11/window.rs

```diff
@@ -0,0 +1,444 @@
+        let visual_type = get_visual_from_screen(&screen).unwrap();
+        let visual_id = visual_type.visual_id();
+
+        let cw_values = [
+            (xcb::CW_BACK_PIXEL, screen.white_pixel()),
+            (
+                xcb::CW_EVENT_MASK,
+                xcb::EVENT_MASK_EXPOSURE
+                    | xcb::EVENT_MASK_KEY_PRESS
+                    | xcb::EVENT_MASK_KEY_RELEASE
+                    | xcb::EVENT_MASK_BUTTON_PRESS
+                    | xcb::EVENT_MASK_BUTTON_RELEASE
+                    | xcb::EVENT_MASK_POINTER_MOTION,
+            ),
+        ];
+
+        // Create the actual window
+        xcb::create_window(
+            // Connection to the X server
+            &conn,
+            // Window depth
+            XCB_COPY_FROM_PARENT.try_into().unwrap(),
+            // The new window's ID
+            window_id,
+            // Parent window of this new window
+            // TODO(#468): either `screen.root()` (no parent window) or pass parent here to attach
+            screen.root(),
+            // X-coordinate of the new window
+            0,
+            // Y-coordinate of the new window
+            0,
+            // Width of the new window
+            // TODO(x11/dpi_scaling): figure out DPI scaling
+            self.size.width as u16,
+            // Height of the new window
+            // TODO(x11/dpi_scaling): figure out DPI scaling
+            self.size.height as u16,
+            // Border width
+            0,
+            // Window class type
+            xcb::WINDOW_CLASS_INPUT_OUTPUT as u16,
+            // Visual ID
+            visual_id,
+            // Window properties mask
+            &cw_values,
+        );
+
+        // Set window title
+        xcb::change_property(
+            &conn,
+            xcb::PROP_MODE_REPLACE as u8,
+            window_id,
+            xcb::ATOM_WM_NAME,
+            xcb::ATOM_STRING,
+            8,
+            self.title.as_bytes(),
+        );
+
+        conn.flush();
+
+        let mut handler = self.handler.unwrap();
+        let handle = WindowHandle::new(window_id);
+        handler.connect(&(handle.clone()).into());
+
+        let xwindow = XWindow::new(window_id, handler, self.size);
+        Application::add_xwindow(window_id, xwindow);
+
+        Ok(handle)
+    }
+}
+
+// X11-specific event handling and window drawing (etc.)
+pub(crate) struct XWindow {
+    window_id: u32,
+    handler: Box<dyn WinHandler>,
+    refresh_rate: Option<f64>,
+}
+
+impl XWindow {
+    pub fn new(window_id: u32, window_handler: Box<dyn WinHandler>, size: Size) -> XWindow {
+        let conn = Application::get_connection();
+        let setup = conn.get_setup();
+        let screen_num = Application::get_screen_num();
+        // TODO(x11/errors): Don't unwrap for screen or visual_type?
+        let screen = setup.roots().nth(screen_num as usize).unwrap();
+        let mut visual_type = get_visual_from_screen(&screen).unwrap();
```

Comment:
This is not used anywhere and the compiler complains about it, I suppose this `visual_type` can go away?

Reply:
Whoops, missed it when I moved this chunk of code to a different function. Deleted (and a few other variables too).